src/App.jsx

```diff
@@ -3,26 +3,61 @@ import './App.scss';
 import './styles/general.scss';
 import { TodoList } from './components/TodoList';
 import { CurrentUser } from './components/CurrentUser';
+import { getAllTodos } from './api';
 
 class App extends React.Component {
   state = {
     todos: [],
     selectedUserId: 0,
   };
 
+  componentDidMount() {
+    getAllTodos()
+      .then(todos => (
+        this.setState({
+          todos: todos.data.filter(task => task.title && task.userId),
+        })));
+  }
+
+  selectUser = (id) => {
+    this.setState({ selectedUserId: id });
+  }
+
+  clearUser = () => {
+    this.setState({ selectedUserId: 0 });
+  }
+
+  changeTodoStatus = (task) => {
+    this.setState(prevState => ({
+      todos: prevState.todos.map((todo) => {
+        if (task === todo) {
+          const newTask = todo;
+
+          newTask.completed = !newTask.completed;
+        }
+
+        return todo;
```

Comment:
You need to return a copy of todo, if it's changed.
const newTask = todo;
newTask.completed = !newTask.completed;
By these lines you don't change anything, it's still changing property of todo, not of its copy